zipline/testing/fixtures.py

```diff
@@ -315,72 +333,218 @@ def init_class_fixtures(cls):
+    """
+    ZiplineTestCase mixin providing cls.bcolz_daily_bar_path,
+    cls.bcolz_daily_bar_ctable, and cls.bcolz_daily_bar_reader class level
+    fixtures.
+
+    After init_class_fixtures has been called:
+    - `cls.bcolz_daily_bar_path` is populated with
+      `cls.tmpdir.getpath(cls.BCOLZ_DAILY_BAR_PATH)`.
+    - `cls.bcolz_daily_bar_ctable` is populated with data returned from
+      `cls.make_daily_bar_data`. By default this calls
+      :func:`zipline.pipeline.loaders.synthetic.make_daily_bar_data`.
+    - `cls.bcolz_daily_bar_reader` is a daily bar reader pointing to the
+      directory that was just written to.
+
+    See Also
+    --------
+    WithBcolzMinuteBarReader
+    WithDataPortal
     """
+    BCOLZ_DAILY_BAR_PATH = 'daily_equity_pricing.bcolz'
+    BCOLZ_DAILY_BAR_LOOKBACK_DAYS = 0
+    BCOLZ_DAILY_BAR_FROM_CSVS = False
+    BCOLZ_DAILY_BAR_USE_FULL_CALENDAR = False
 
-    BCOLZ_MINUTES_PER_DAY = US_EQUITIES_MINUTES_PER_DAY
+    @classmethod
+    def make_daily_bar_data(cls):
+        return create_daily_bar_data(
+            cls.bcolz_daily_bar_days,
+            cls.asset_finder.sids,
+        )
 
     @classmethod
     def init_class_fixtures(cls):
-        super(WithBcolzMinutes, cls).init_class_fixtures()
+        super(WithBcolzDailyBarReader, cls).init_class_fixtures()
+        cls.bcolz_daily_bar_path = p = cls.tmpdir.makedir(
+            cls.BCOLZ_DAILY_BAR_PATH,
+        )
+        cls.bcolz_daily_bar_days = days = (
+            cls.env.trading_days
+            if cls.BCOLZ_DAILY_BAR_USE_FULL_CALENDAR else
+            cls.env.days_in_range(
+                cls.env.trading_days[
+                    cls.env.get_index(cls.sim_params.period_start) -
+                    cls.BCOLZ_DAILY_BAR_LOOKBACK_DAYS
+                ],
+                cls.sim_params.period_end,
+            )
+        )
+        cls.bcolz_daily_bar_ctable = t = getattr(
+            BcolzDailyBarWriter(p, days),
+            'write_csvs' if cls.BCOLZ_DAILY_BAR_FROM_CSVS else 'write',
+        )(cls.make_daily_bar_data())
+
+        cls.bcolz_daily_bar_reader = BcolzDailyBarReader(t)
+
+
+class WithBcolzMinuteBarReader(WithTmpDir, WithSimParams):
+    """
+    ZiplineTestCase mixin providing cls.bcolz_minute_bar_path,
+    cls.bcolz_minute_bar_ctable, and cls.bcolz_minute_bar_reader class level
+    fixtures.
+
+    After init_class_fixtures has been called:
+    - `cls.bcolz_minute_bar_path` is populated with
+      `cls.tmpdir.getpath(cls.BCOLZ_MINUTE_BAR_PATH)`.
+    - `cls.bcolz_minute_bar_ctable` is populated with data returned from
+      `cls.make_minute_bar_data`. By default this calls
+      :func:`zipline.pipeline.loaders.synthetic.make_minute_bar_data`.
+    - `cls.bcolz_minute_bar_reader` is a minute bar reader pointing to the
+      directory that was just written to.
 
+    See Also
+    --------
+    WithBcolzDailyBarReader
+    WithDataPortal
+    """
+    BCOLZ_MINUTE_BAR_PATH = 'minute_equity_pricing.bcolz'
```

Comment:
also, I don't think any tests currently configure this but I think it is nice to have control over the shared tmpdir for a suite